packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -643,60 +644,146 @@ export class SDKClient {
+      let transactionFee = createFileRecord.transactionFee;
+      this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
 
       this.captureMetrics(
         SDKClient.transactionMode,
-        fileAppendTx.constructor.name,
+        fileCreateTx.constructor.name,
         Status.Success,
-        await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
-        0,
+        createFileRecord.transactionFee.toTinybars().toNumber(),
+        createFileRecord?.contractFunctionResult?.gasUsed,
         callerName,
         interactingEntity,
       );
-    }
 
-    // Ensure that the calldata file is not empty
-    if (fileId) {
-      const fileSize = await (await new FileInfoQuery().setFileId(fileId).execute(client)).size;
+      if (fileId && callData.length > this.fileAppendChunkSize) {
+        const fileAppendTx = await new FileAppendTransaction()
+          .setFileId(fileId)
+          .setContents(hexedCallData.substring(this.fileAppendChunkSize, hexedCallData.length))
+          .setChunkSize(this.fileAppendChunkSize)
+          .setMaxChunks(this.maxChunks);
+        const fileAppendTxResponse = await fileAppendTx.execute(client);
+
+        // get transaction fee and add expense to limiter
+        const appendFileRecord = await fileAppendTxResponse.getRecord(this.clientMain);
+        transactionFee = appendFileRecord.transactionFee;
+        this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
+
+        this.captureMetrics(
+          SDKClient.transactionMode,
+          fileAppendTx.constructor.name,
+          Status.Success,
+          await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
+          0,
+          callerName,
+          interactingEntity,
+        );
+      }
+
+      // Ensure that the calldata file is not empty
+      if (fileId) {
+        const fileSize = await (await new FileInfoQuery().setFileId(fileId).execute(client)).size;
 
-      if (callData.length > 0 && fileSize.isZero()) {
-        throw new SDKClientError({}, `${requestIdPrefix} Created file is empty. `);
+        if (callData.length > 0 && fileSize.isZero()) {
+          throw new SDKClientError({}, `${requestIdPrefix} Created file is empty. `);
+        }
+        this.logger.trace(`${requestIdPrefix} Created file with fileId: ${fileId} and file size ${fileSize}`);
       }
-      this.logger.trace(`${requestIdPrefix} Created file with fileId: ${fileId} and file size ${fileSize}`);
-    }
 
-    return fileId;
+      return fileId;
+    } catch (error: any) {
+      const sdkClientError = new SDKClientError(error, error.message);
+      let transactionFee: number | Hbar = 0;
+
+      // if valid network error utilize transaction id
+      if (sdkClientError.isValidNetworkError()) {
+        try {
+          const transactionCreateRecord = await new TransactionRecordQuery()
+            .setTransactionId(fileCreateTx.transactionId!)
```

Comment:
This `fileCreateTx` is always null as initialized with no value at [line 654](https://github.com/hashgraph/hedera-json-rpc-relay/pull/2634/files#diff-4b0bcc39d9efb3e923f041c7459f1cc42d34597d4200839df6764c2a345a1c1bR654)

Reply:
Applied suggested fix which resolves this